### DISCUSSION TITLE
53 port is used

### DISCUSSION THREAD

**Discussion by PadowYT2** [Seji64/SniDust#70] [Q&A] [answered]
I'm trying to setup SniDust, but port 53 is already used by systemd-resolved. When I disable & stop it, SniDust starts, but I got no internet on the machine

Also I was wondering if I can pair this with https://github.com/bol-van/zapret

**Comment by Seji64** [comment, marked as answer]
> I'm trying to setup SniDust, but port 53 is already used by systemd-resolved. When I disable & stop it, SniDust starts, but I got no internet on the machine

https://github.com/Seji64/SniDust#error-port-53-is-already-in-use

> Also I was wondering if I can pair this with https://github.com/bol-van/zapret

No clue, but i don't think you can combine those two tools.

**Reply by PadowYT2** [reply]
I'll check out about the port, about zapret. It will install as a system process, so it should work...? If it really won't work. Is it possible to attach a proxy?

**Reply by Seji64** [reply]
I'm not quite sure what you're trying to achieve?
Is snidust supposed to use zapret or the client?

**Reply by PadowYT2** [reply]
> I'm not quite sure what you're trying to achieve?
> Is snidust supposed to use zapret or the client?

I think I didn't think this through. Basically, I installed zapret on the server for DPI bypass, then using SniDust, having zapret running aswell on the proxy, so if I use the DNS server, zapret would work magically (how it should)

**Comment by wolojonathan** [comment]
Hi,

If your question is "Will it be compatible if you are using both tools simultaneously ?", the short answer is : Yes, SniDust will work with DNS queries performed from a source behind a zapret proxy.
I myself am using both zapret (Tried both tpws and nfqws) and SniDust on a remote server - It works like a charm!

However, I'll detail a bit and give a comprehensive answer, as zapret and SniDust are two totally different tools.

In case you are trying to bypass DPI, zapret, depending on its configuration, will alter the transmitted packets FROM your location (your home usually, for example by disordering them, modifying the headers case, etc ...), and it must happen before the packets are reaching your ISP.
Usually, zapret is installed on your router or on your home server for that.

SniDust however is altering the DNS queries in order to spoof them as if they were executed from the VPS location. Logically, that would be another geographical location than the one you are really executing the query from.

The way would be : Home -> zapret (DPI bypass) -> ISP Origin -> Remote server -> SniDust -> Target resource

Therefore, I agree with @Seji64 and it seems to me a bit strange to have both tools on the same machine :)
